cmd/syncthing/gui.go

```diff
@@ -1024,6 +1027,96 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileEntry struct {
+	name string
+	data []byte
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var arrayFile []fileEntry
+
+	// Redacted configuration as a JSON
+	jsonConfig, _ := json.MarshalIndent(getRedactedConfig(s), "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "config.json", data: jsonConfig})
+
+	// Log as a text
+	var buff bytes.Buffer
+	for _, line := range s.systemLog.Since(time.Time{}) {
+		fmt.Fprintf(&buff, "%s: %s\n", line.When.Format(time.RFC3339), line.Message)
+	}
+	arrayFile = append(arrayFile, fileEntry{name: "log.txt", data: buff.Bytes()})
+
+	// Errors as a JSON
+	if errs := s.guiErrors.Since(time.Time{}); len(errs) > 0 {
+		jsonError, _ := json.MarshalIndent(errs, "", "  ")
+		arrayFile = append(arrayFile, fileEntry{name: "errors.json", data: jsonError})
+	}
+
+	// Errors as a JSON
+	panicFiles, ok := checkPanicFile(baseDirs["config"])
+	if ok {
+		for _, f := range panicFiles {
+			panicF, err := ioutil.ReadFile(f)
+			if err != nil {
+				arrayFile = append(arrayFile, fileEntry{name: filepath.Base(f), data: panicF})
+			}
+		}
+	}
+
+	// Version and platform information as a JSON
+	versionPlatform, _ := json.MarshalIndent(map[string]string{
+		"version":     Version,
+		"codename":    Codename,
+		"longVersion": LongVersion,
+		"os":          runtime.GOOS,
+		"arch":        runtime.GOARCH,
+	}, "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "versionPlatform.json", data: versionPlatform})
+
+	// Report Data as a JSON
+	version := usageReportVersion
+	if val, _ := strconv.Atoi(r.URL.Query().Get("version")); val > 0 {
```

Comment:
I think skip this. On the regular usage reporting endpoint we need to be able to select usage reporting version, but here's that's not relevant. You can just pass the `usageReportVersion` constant directly to `reportData` to get the latest version.